libtransmission/utils.c

```diff
@@ -456,6 +457,31 @@ char const* tr_memmem(char const* haystack, size_t haystacklen, char const* need
 #endif
 }
 
+char const* tr_memmemcase(char const* haystack, size_t haystacklen, char const* needle, size_t needlelen)
+{
+
+    if (needlelen == 0)
+    {
+        return haystack;
+    }
+
+    if (needlelen > haystacklen || haystack == NULL || needle == NULL)
+    {
+        return NULL;
+    }
+
+    for (size_t i = 0; i <= haystacklen - needlelen; ++i)
+    {
+        if (evutil_ascii_strncasecmp(haystack + i, needle, needlelen) == 0)
+        {
+            return haystack + i;
+        }
+    }
+
+    return NULL;
+
+}
```

Comment:
Looks like `StrStrA()` does this on Windows and `strcasestr()` does it everywhere else.

I'd much rather use system functions when available, since they're more likely to be bulletproof  & fast. This implementation would work but is pretty brute-force :slightly_smiling_face: 

Maybe we should have a `tr_strcasestr()` that's implemented with platform-specific #ifdefs similar to, say, `tr_localtime_r()`, e.g.:

```c
#if defined(HAVE_STRCASESTR)
  return strcasestr(haystack, needle);
#elif defined(_WIN32)
  return StrStrA(haystack, needle);
#else 
  #error please open a PR to implement tr_strcasestr() for your platform
#endif 
```

(TODO: add HAVE_STRCASESTR to cmake/autogen build tooling)

Reply:
@ckerr thank you for your review, I also thought this was a bit of a hack but I was not too familiar with standard C functions so I had no idea that existed. I can add these changes but can you point me in the right direction on where I would need to add this check in cmake/autogen? I have no experience in either of them.

Reply:
Sure! Look in `libtransmission/utils.h` and look at the way `#ifdef` is used there, e.g. `#ifdef HAVE_STRSEP, HAVE_STRLCPY, HAVE_ICONV, etc etc. and then look at how those get defined in the cmake and autogen files. The code for HAVE_STRCASESTR would be pretty similar.

Reply:
Thank you, I will try something this week when I have some free time.